storage/Devices/PartitionImpl.cc

```diff
@@ -269,6 +307,34 @@ namespace storage
 
 
     void
+    Partition::Impl::set_boot(bool boot)
+    {
+	if (!is_msdos(get_partition_table()))
+	    ST_THROW(Exception("set_boot only supported for Msdos"));
+
+	if (boot && !Impl::boot)
+	{
+	    PartitionTable* partition_table = get_partition_table();
+
+	    for (Partition* partition : partition_table->get_partitions())
```

Comment:
Well, then those programs are not standard-conformant: "Standard-conformant MBRs will allow only one partition marked active and use this as part of a sanity-check to determine the existence of a valid partition table." (https://en.wikipedia.org/wiki/Master_boot_record)

But I can add a sentence about this in the documentation.